integration/kubernetes/confidential/image_pulling_with_snapshotter.bats

```diff
@@ -0,0 +1,155 @@
+
+    echo "Prepare containerd for Confidential Container"
+    SAVED_CONTAINERD_CONF_FILE="/etc/containerd/config.toml.$$"
+    configure_cc_containerd "$SAVED_CONTAINERD_CONF_FILE"
+
+    echo "Reconfigure Kata Containers"
+    switch_image_service_offload off
+    clear_kernel_params
+    add_kernel_params "${original_kernel_params}"
+
+    setup_proxy
+    switch_measured_rootfs_verity_scheme none
+}
+
+# Check the logged messages on host have a given message.
+# Parameters:
+#      $1 - the message
+#
+# Note: get the logs since the global $start_date.
+#
+assert_logs_contain() {
+    local message="$1"
+    # Note: with image-rs we get more that the default 1000 lines of logs
+    journalctl -x -t kata --since "$start_date" -n 100000 | grep "$message"
+}
+
+@test "$test_tag Test can pull an image as a raw block disk image to guest with dm-verity enabled" {
+    if [ "$SNAPSHOTTER" = "nydus" ]; then
+        EXPORT_MODE="image_block_with_verity" RUNTIMECLASS="$RUNTIMECLASS" SNAPSHOTTER="nydus" configure_remote_snapshotter
+        pod_config="$(new_pod_config "$image_unsigned_unprotected")"
+        echo $pod_config
+        create_test_pod "$pod_config"
+        remove_test_image "$image_unsigned_unprotected"
+    fi
+}
+
+@test "$test_tag Test can create two pods with pulling the image only once" {
+    if [ "$SNAPSHOTTER" = "nydus" ]; then
+        EXPORT_MODE="image_block_with_verity" RUNTIMECLASS="$RUNTIMECLASS" SNAPSHOTTER="nydus" configure_remote_snapshotter
+
+        pod_config_1="$(new_pod_config "$image_unsigned_unprotected" "1")"
+        echo $pod_config_1
+        create_test_pod $pod_config_1
+        pod_config_2="$(new_pod_config "$image_unsigned_unprotected" "2")"
+        echo $pod_config_2
+        create_test_pod $pod_config_2
+
+        pull_times=$(journalctl -g "PullImage \"$image_unsigned_protected\" with snapshotter nydus" | wc -l)
+        [ ${#pull_times[@]} -eq 1 ]
+        remove_test_image "$image_unsigned_unprotected"
+    fi
+}
+
+@test "$test_tag Test can pull an image inside the guest with remote-snapshotter" {
+    switch_image_service_offload on
+    if [ "$SNAPSHOTTER" = "nydus" ]; then
+        EXPORT_MODE="image_guest_pull" RUNTIMECLASS="$RUNTIMECLASS" SNAPSHOTTER="nydus" configure_remote_snapshotter
+        create_test_pod
+        remove_test_image "$image_unsigned_unprotected"
+    fi
+}
+
+teardown() {
+    # Print the logs and cleanup resources.
+    echo "-- Kata logs:"
+    sudo journalctl -xe -t kata --since "$start_date" -n 100000
+
+    # Allow to not destroy the environment if you are developing/debugging
+    # tests.
+    if [[ "${CI:-false}" == "false" && "${DEBUG:-}" == true ]]; then
+        echo "Leaving changes and created resources untouched"
+        return
+    fi
+
+    kubernetes_delete_all_cc_pods_if_any_exists || true
+    clear_kernel_params
+    add_kernel_params "${original_kernel_params}"
+    switch_image_service_offload off
+    remove_nydus_snapshotter_from_containerd
```

Comment:
I wonder if we should use the model that the cri test use where, rather than updating the containerd config with `remove_nydus_snapshotter_from_containerd`, which could be error prone, we just save the start state (that `configure_cc_containerd` already does) and restore it at the end of the test like:

https://github.com/kata-containers/tests/blob/6e28155426202bebf85c591536fa7a82b4589b5e/integration/containerd/confidential/tests_common.sh#L90-L96

Reply:
Sure. I'll update.

Reply:
Fixed.